src/Microsoft.AspNet.Mvc.Core/ParameterBinding/ModelBindingHelper.cs

```diff
@@ -147,9 +147,7 @@ public static class ModelBindingHelper
                [NotNull] Func<ModelBindingContext, string, bool> predicate)
            where TModel : class
         {
-            var modelMetadata = metadataProvider.GetMetadataForType(
-                modelAccessor: () => model,
-                modelType: model.GetType());
+            var modelMetadata = metadataProvider.GetMetadataForType(model?.GetType() ?? typeof(TModel));
```

Comment:
did this code `throw` in the past?  the two `null` checks are not needed if `model != null` is an invariant here.